File: Python/8py 1/find_needle.py
"""
|--------------------------------------------------------------------------
| A Needle in the Haystack
|--------------------------------------------------------------------------
|
"""

def find_needle(haystack):
   return "found the needle at position "+str(haystack.index("needle"))


find_needle = lambda haystack: "found the needle at position {}".format(haystack.index("needle"))


def find_needle(haystack):
	return f'found the needle at position {haystack.index("needle")}'


def find_needle(haystack):
	index = haystack.index('needle')
	return 'found the needle at position {}'.format(index)


def find_needle(haystack):
	return 'found the needle at position %s' % haystack.index('needle')


print(find_needle(['3', '123124234', None, 'needle', 'world', 'hay', 2, '3', True, False]), 'found the needle at position 3')
print(find_needle(['283497238987234', 'a dog', 'a cat', 'some random junk', 'a piece of hay', 'needle', 'something somebody lost a while ago']), 'found the needle at position 5')
print(find_needle([1,2,3,4,5,6,7,8,8,7,5,4,3,4,5,6,67,5,5,3,3,4,2,34,234,23,4,234,324,324,'needle',1,2,3,4,5,5,6,5,4,32,3,45,54]), 'found the needle at position 30')

File: PHP/8php 1/findNeedle.php
<?php
/*
|--------------------------------------------------------------------------
| A Needle in the Haystack
|--------------------------------------------------------------------------
|
*/

function findNeedle($haystack) {
   for ($i=0; $i < count($haystack); $i++) { 
      if ($haystack[$i] == 'needle') {
         return "found the needle at position {$i}";
      }
   }
}


function findNeedle($a) {
   return "found the needle at position " . array_search("needle", $a);
}


function findNeedle($haystack) {
   $haystack = array_map('strval', $haystack);
   $flipped = array_flip($haystack);
   return 'found the needle at position ' . $flipped['needle'];
}


function findNeedle($h) {
   $i=0;
   while ($h[$i] <> 'needle') { 
      $i+=1; 
   }
   return 'found the needle at position '.$i;
}


$haystack_1 = ['3', '123124234', null, 'needle', 'world', 'hay', 2, '3', true, false];
$haystack_2 = ['283497238987234', 'a dog', 'a cat', 'some random junk', 'a piece of hay', 'needle', 'something somebody lost a while ago'];
print_r(findNeedle($haystack_1). ' found the needle at position 3
');
print_r(findNeedle($haystack_2). ' found the needle at position 5
');